DLSv2/Core/ManagedVehicle.cs

```diff
@@ -127,16 +125,15 @@ public bool LightsOn
         /// </summary>
         public bool SirenOn { get; set; } = false;
         public Dictionary<string, int> SoundIds = new Dictionary<string, int>();
-        public Dictionary<string, (bool IsActive, int Index)> AudioControlGroups = new Dictionary<string, (bool, int)>();
-        public Dictionary<string, (bool IsManualing, int Index)> AudioCGManualing = new Dictionary<string, (bool, int)>();
-        public Dictionary<string, bool> AudioModes = new Dictionary<string, bool>();
-        public List<string> ActiveAudioModes = new List<string>();
+        public Dictionary<string, ControlGroupInstance<AudioControlGroup>> AudioControlGroups = new();
+        public Dictionary<string, ModeInstance<AudioMode>> AudioModes = new();
 
         // Registers input
         public void RegisterInputs()
         {
+            ControlsManager.ClearInputs();
             // Light Inputs
-            foreach (ControlGroup cG in ControlGroupManager.ControlGroups[Vehicle.Model].Values)
+            foreach (ControlGroup cG in LightControlGroups.Values.Select(x => x.BaseControlGroup))
```

Comment:
I would change this to do `foreach(ControlGroupWrapper cgw in LightControlGroups.Values)`. You can get the base control group where you need it, but this would allow you to avoid all of the repeated use of `LightControlGroups[cg.Name].Toggle()` etc. and just directly use `cgw.Toggle()`. You're basically getting the items from a dictionary, looping through just the name of them, then getting them back from the same dictionary, which is inefficient when you could just use the items directly.